source/egg/math/Quat.cc

```diff
@@ -35,4 +43,39 @@ Vector3f Quatf::rotateVector(const Vector3f &vec) const {
     return res.v;
 }
 
+Quatf Quatf::slerpTo(const Quatf &q1, f32 t) const {
+    f32 dot_ = std::max(-1.0f, std::min(1.0f, dot(q1)));
+    bool bDot = dot_ < 0.0f;
+    dot_ = std::abs(dot_);
+
+    f32 acos = Mathf::acos(dot_);
+    f32 sin = Mathf::sin(acos);
+
+    f32 s;
+    if (std::abs(sin) < 0.00001f) {
+        s = 1.0f - t;
+    } else {
+        f32 invSin = 1.0f / sin;
+        f32 tmp0 = t * acos;
+        s = invSin * Mathf::sin(acos - tmp0);
+        t = invSin * Mathf::sin(tmp0);
+    }
+
+    if (bDot) {
+        t = -t;
+    }
+
+    return Quatf(s * w + t * q1.w, s * v + t * q1.v);
+}
+
+f32 Quatf::dot() const {
+    return w * w + v.dot();
+}
+
+f32 Quatf::dot(const Quatf &q) const {
+    return w * q.w + v.dot(q.v);
+}
```

Comment:
These functions don't exist in EGG. This invokes the question of whether or not we care.

I think it's fine to introduce these (and subsequently, the matched `proj` and `rej` methods), but I'm curious to hear more thoughts.

Reply:
I personally don't see any benefit to matching the game in instances like these. I think it's appropriate to add functions like these where applicable.

Reply:
True. Sounds good to me.